VMware: Fix KeyError in vmware_host_config_manager

mariolenz opened the pull request:
##### SUMMARY
option_manager.supportedOption may return options that aren't set which results in a KeyError

Fixes #44561

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
vmware_host_config_manager

##### ADDITIONAL INFORMATION
The problem is that _first_ all options that are set are retrieved and _then_ this list is updated with every option's type. When 'Vpx.Vpxa.config.log.level' is supported but not set, this results in a KeyError because `host_facts[Vpx.Vpxa.config.log.level']` doesn't exist.

I think it is better to _first_ get a list of all supported options and _then_ update this list with every option's actual value.

mariolenz committed:
Bug fix for vmware_host_config_manager

Akasurde commented:
@mariolenz Thanks for the PR. Could you please add a changelog fragment like [https://github.com/ansible/ansible/pull/52642/files#diff-67cd9196cfa0b7f27a27b45181d8c5ab] ?

mariolenz commented:
> @mariolenz Thanks for the PR. Could you please add a changelog fragment like [this](https://github.com/ansible/ansible/pull/52642/files#diff-67cd9196cfa0b7f27a27b45181d8c5ab) ?

I've added a changelog fragment and squashed the commit with the previous one because I didn't want to clutter your commit history. Hope it's OK now, otherwise just tell me :-)

Akasurde commented on changelogs/fragments/44561-vmware_host_config_manager-fix_key_error.yml:
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+- Fixed issue when a supported option isn't already set (https://github.com/ansible/ansible/issues/44561).
```
```suggestion
- Fixed KeyError issue in vmware_host_config_manager when a supported option isn't already set (https://github.com/ansible/ansible/issues/44561).
```

Akasurde replied:
Since changelog entries will be combined together in final release version, there will be no clue about this line.

Akasurde committed:
Update changelogs/fragments/44561-vmware_host_config_manager-fix_key_error.yml

Co-Authored-By: mariolenz <m@riolenz.de>

Akasurde reviewed (approved):
LGTM

Akasurde commented:
@pdellaert @dericcrago @Im0 @ckotte @jeking3 @Tomorrow9 Could you please review this ?

pdellaert reviewed (approved):
LGTM

Akasurde commented:
@mariolenz Thanks for the contribution. @pdellaert Thanks for the review.
